server/command_hooks.go

```diff
@@ -722,3 +762,89 @@ func (p *Plugin) executeAutocompleteTest(args *model.CommandArgs) *model.Command
 		Text:         fmt.Sprintf("Executed command: " + args.Command),
 	}
 }
+
+func (p *Plugin) executeCommandPreferences(args *model.CommandArgs) *model.CommandResponse {
+	fields := strings.Fields(args.Command)
+	command := ""
+	if len(fields) == 2 {
+		command = fields[1]
+	}
+
+	switch command {
+	case help:
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         commandPreferencesHelp,
+		}
+	case "get":
+		var name string
+		if len(fields) != 3 {
+			return &model.CommandResponse{
+				ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+				Text:         "Please provide preference name",
+			}
+		}
+		name = fields[2]
+		prefs, err := p.API.GetPreferencesForUser(args.UserId)
+		if err != nil {
+			p.API.LogError("Failed to get user preferences", "err", err.Error())
+			return &model.CommandResponse{}
+		}
+		for _, p := range prefs {
+			if p.Name == name {
```

Comment:
From looking at the content of `prefs' it doesn't seem like we can easily match on `Name`:
```
{
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "tutorial_step", Name: "iqbqga6mdjfa3prh59er8fh57c", Value: "999"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_approximate_view_time", Name: "", Value: "1605693815487"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_open_time", Name: "m93njr5obbyqirzf6raixu63hw", Value: "1604692117885"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "direct_channel_show", Name: "edi3ewpbcpru5gftmpxrhna31o", Value: "true"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_approximate_view_time", Name: "xnu1rgdpiffwjfb15sgdnhuaho", Value: "1604314238495"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "theme", Name: "", Value: "{\"awayIndicator\":\"#ffbc42\",\"buttonBg\":\"#166de0\",\"buttonColor\":\"#ffffff\",\"centerChannelBg\":\"#ffffff\",\"centerChannelColor\":\"#3d3c40\",\"codeTheme\":\"github\",\"dndIndicator\":\"#f74343\",\"errorTextColor\":\"#fd5960\",\"image\":\"/static/files/460473bcf310d651a1d68e35e484df97.png\",\"linkColor\":\"#2389d7\",\"mentionBg\":\"#ffffff\",\"mentionColor\":\"#145dbf\",\"mentionHighlightBg\":\"#ffe577\",\"mentionHighlightLink\":\"#166de0\",\"newMessageSeparator\":\"#ff8800\",\"onlineIndicator\":\"#06d6a0\",\"sidebarBg\":\"#145dbf\",\"sidebarHeaderBg\":\"#1153ab\",\"sidebarHeaderTextColor\":\"#ffffff\",\"sidebarText\":\"#ffffff\",\"sidebarTextActiveBorder\":\"#579eff\",\"sidebarTextActiveColor\":\"#ffffff\",\"sidebarTextHoverBg\":\"#4578bf\",\"sidebarUnreadText\":\"#ffffff\",\"type\":\"Mattermost\"}"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "direct_channel_show", Name: "h7qmwf3d7tb9mqtwzpa7njgapr", Value: "true"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_open_time", Name: "uxhczcwh3pfytj9dh5fu6bhp1h", Value: "1604575010094"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "recommended_next_steps", Name: "skip", Value: "true"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "direct_channel_show", Name: "iqbqga6mdjfa3prh59er8fh57c", Value: "true"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_open_time", Name: "umwssh8uy7fcdxoghsgwcst9we", Value: "1604496816318"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_approximate_view_time", Name: "son1npofqidwxjkodkrjbb44ua", Value: "1604400840350"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "direct_channel_show", Name: "g64gf5qtk7dp7cnxcy1p8k9g4o", Value: "true"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_open_time", Name: "kjkbt4z8kfbuuyt7h1qmy7ebuy", Value: "1604056605335"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_open_time", Name: "1i9xpftodjbn3mspx3zgnjj6gc", Value: "1604918656936"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "direct_channel_show", Name: "8se64adhfb8kdkbc9up6rqi5cc", Value: "true"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_open_time", Name: "rf76s354mbg73mhsjpcnikzyew", Value: "1604563673011"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "direct_channel_show", Name: "i35936p3cty7jyys88jidycmfh", Value: "true"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "channel_open_time", Name: "pmpc15r9f3yh5pk6ad4t9bwpsw", Value: "1604496817135"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "recommended_next_steps", Name: "hide", Value: "true"},
    model.Preference{UserId: "iqbqga6mdjfa3prh59er8fh57c", Category: "direct_channel_show", Name: "5k91qhwi9fbixq77cknfm3ywia", Value: "true"}
}
```

Could you please investigate if there is a way to map the preference to a human readable text?

Reply:
I think category could be the actual name for preference because Name is just a UUID. Need to see how it is set across the code.

Reply:
@hmhealey Maybe you can help here. Do you know what kann be used as a human readable text to set/get specific preferences?

Reply:
The category/name are both arbitrary strings, so some are human readable while others aren't. For ones with the category `direct_channel_show`, the name is the user ID if the other user in the DM, but for ones with the category `sidebar_settings`, the name will be something readable like `show_unread_section`. Generally, you'd want both the category and name to address a preference, but the category is there to group related preferences in the case where you want something like all DMs with visible channels in the sidebar.